src/phoenix/server/api/schema.py

```diff
@@ -384,6 +402,140 @@ async def create_dataset(
+            )
+            input_value = _get_attribute_value(models.Span.attributes, INPUT_VALUE).label(
+                "input_value"
+            )
+            output_mime_type = _get_attribute_value(models.Span.attributes, OUTPUT_MIME_TYPE).label(
+                "output_mime_type"
+            )
+            output_value = _get_attribute_value(models.Span.attributes, OUTPUT_VALUE).label(
+                "output_value"
+            )
+            llm_prompt_template_variables = _get_attribute_value(
+                models.Span.attributes, LLM_PROMPT_TEMPLATE_VARIABLES
+            ).label("llm_prompt_template_variables")
+            llm_input_messages = _get_attribute_value(
+                models.Span.attributes, LLM_INPUT_MESSAGES
+            ).label("llm_input_messages")
+            llm_output_messages = _get_attribute_value(
+                models.Span.attributes, LLM_OUTPUT_MESSAGES
+            ).label("llm_output_messages")
+            retrieval_documents = _get_attribute_value(
+                models.Span.attributes, RETRIEVAL_DOCUMENTS
+            ).label("retrieval_documents")
+            span_query_result = await session.execute(
+                select(
+                    models.Span.id,
+                    models.Span.span_kind,
+                    models.Span.attributes,
+                    input_mime_type,
+                    input_value,
+                    output_mime_type,
+                    output_value,
+                    llm_prompt_template_variables,
+                    llm_input_messages,
+                    llm_output_messages,
+                    retrieval_documents,
+                )
+                .select_from(models.Span)
+                .where(models.Span.id.in_(span_rowids))
+            )
+            spans = span_query_result.all()
+            if missing_span_rowids := span_rowids - {span.id for span in spans}:
+                raise ValueError(
+                    f"Could not find spans with rowids: {", ".join(map(str, missing_span_rowids))}"
+                )  # todo: implement error handling types https://github.com/Arize-ai/phoenix/issues/3221
+            for span in spans:
+                dataset_example_rowid = await session.scalar(
+                    insert(models.DatasetExample)
+                    .values(dataset_id=dataset_rowid, span_rowid=span.id)
+                    .returning(models.DatasetExample.id)
+                )
+                await session.execute(
+                    insert(models.DatasetExampleRevision).values(
+                        dataset_example_id=dataset_example_rowid,
+                        dataset_version_id=dataset_version_rowid,
+                        input=get_dataset_example_input(span),
+                        output=get_dataset_example_output(span),
+                        metadata_=span.attributes,
+                        revision_kind="CREATE",
+                    )
+                )
```

Comment:
Is there a more efficiently way of doing this query in SQLAlchemy?

Reply:
Going to switch to [this pattern](https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#bulk-insert-with-returning).

Reply:
addressed in [f72d4ee](https://github.com/Arize-ai/phoenix/pull/3205/commits/f72d4ee537434076c0495e8bbe845dabbc49a439)